### PR TITLE
chore(geoip transform): Remove the deprecated `geoip` transform

### PR DESCRIPTION
**Pull request by fuchsnj**
The `geoip` transform was deprecated in `0.24.0` and is now gone.

### CODE REVIEW

File: src/transforms/dedupe.rs [hunk 1]
```diff
@@ -97,6 +97,11 @@ fn default_cache_config() -> CacheConfig {
 // configuration schema _without_ actually changing how a field is populated during deserialization.
```
Comment (spencergilbert) [inline src/transforms/dedupe.rs:97]: Accidentally included?

**Review by spencergilbert — APPROVED** [review]

**Comment by spencergilbert** [comment]
I wouldn't be surprised if this fails netlify building by breaking a link 👀

**Review by tobz — CHANGES_REQUESTED** [review]
Can you also remove the "base" version of the Cue documentation file? Should be at `website/cue/reference/components/transforms/base/geoip.cue`.

**Review by spencergilbert — CHANGES_REQUESTED** [review]
https://github.com/vectordotdev/vector/blob/master/website/content/en/docs/reference/configuration/transforms/geoip.md

This needs to be deleted as well, and this almost certainly will break some links...

File: .github/semantic.yml [hunk 2]
```diff
@@ -166,7 +166,6 @@ scopes:
   - aws_ec2_metadata transform # Anything `aws_ec2_metadata` transform related
   - dedupe transform # Anything `dedupe` transform related
   - filter transform # Anything `filter` transform related
-  - geoip transform # Anything `geoip` transform related
```
Comment (spencergilbert) [inline .github/semantic.yml:169]: noting that this is produced via terraform in the resources repo

Comment (fuchsnj) [reply .github/semantic.yml:169]: https://github.com/vectordotdev/resources/pull/177

**Review by tobz — APPROVED** [review]

**Pull request merged**